TECHNICAL_OVERVIEW.md

```diff
@@ -0,0 +1,362 @@
+| Deniability                                | N       | TBD     | Prevent the recipient of a message from proving to an outsider that a message originated from the sender (as opposed to being forged by the recipient) |
+| Message authentication                     | Y       | Y       | Prevent attackers from masquerading as a conversation participant                                                                                      |
+| Message confidentiality                    | Y       | Y       | Prevent attackers from reading the contents of each message                                                                                            |
+| Message integrity                          | Y       | Y       | Prevent attackers from tampering with the contents of each message                                                                                     |
+
+**Why Olm?**
+
+- Olm is actively used in production and its open source implementations have been tested and [independently audited](https://matrix.org/media/Least%20Authority%20-%20Matrix%20vodozemac%20Final%20Audit%20Report.pdf), which is a signal of reliability.
+- The Olm protocol and implementations such as [vodozemac](https://github.com/matrix-org/vodozemac) are actively maintained and updated, and continue to benefit from security improvements and updates over time.
+- Olm implementations such as vodozemac carry an Apache license, allowing third party organizations to use the XMTP SDK in closed source applications. Additionally, XMTP may make modifications to it, or replace it, in the future as needed.
+
+Using an existing library like Olm for the first iteration of double ratchet in XMTP can reduce the risks associated with "rolling your own" cryptographic implementation. Limitations of Olm are described in [Limitations and future work](#limitations-and-future-work).
+
+## The messaging protocol
+
+In XMTP v3, **conversations** between **accounts** are carried out via pair-wise **sessions** between app **installations**:
+
+- An **account** consists of multiple app installations. To learn more, see [Account management](#account-management).
+- A **conversation** consists of multiple pair-wise sessions between installations. To learn more, see [Conversation management](#conversation-management).
+- Messages are individually encrypted and delivered at the **session** level. To learn more, see [Session management](#session-management).
+- Delivery of messages happens via network **topics**. To learn more, see [Network topic structure](#network-topic-structure).
+
+### Network topic structure
+
+XMTP nodes currently store data using Waku topics. Clients may publish, request, or subscribe to data from these topics. The purpose of these topics will be described in subsequent sections.
+
+- **`contact-<walletAddress>`**: Signed installation contact bundles for a given user account.
```

Comment:
Is this going to interfere with v2?

Reply:
We use a different prefix for all v3 topics so there is no potential for collision.